src/codelabs/iterables.md

```diff
@@ -0,0 +1,608 @@
+
+When working with `Iterable`, sometimes you need to verify that the elements of a collection
+satisfy some condition.
+
+You might be tempted to write a solution using a `for-in` loop like this one:
+
+{:.bad}
+<?code-excerpt "iterables/test/iterables_test.dart (every-bad)"?>
+{% prettify dart %}
+for (var item in items) {
+  if (item.length < 5) {
+    return false;
+  }
+}
+return true;
+{% endprettify %}
+
+However, you can accomplish the same using the `every()` method:
+
+<?code-excerpt "iterables/test/iterables_test.dart (every-good)"?>
+{% prettify dart %}
+return items.every((element) => element.length >= 5);
+{% endprettify %}
+
+Using the `every()` method results in code that is more readable, compact, and less error prone.
+
+### Example: Using any() and every()
+
+The `Iterable` class provides two methods that you can use to verify conditions:
+
+* `any()`: Returns true if at least one element satisfies the condition.
+* `every()`: Returns true if all elements satisfy the condition.
+
+Run this exercise to see them in action.
+
+[//]: https://gist.github.com/d56963729339cea951e16209e0a26e4c
+<iframe
+  src="{{site.dartpad-embed}}?id=d56963729339cea951e16209e0a26e4c"
+  style="border: 1px solid lightgrey; margin-top: 10px; margin-bottom: 25px"
+  frameborder="no"
+  height="375"
+  width="100%" >
+</iframe>
+
+In the example, `any()` verifies that at least one element contains the
+character `a`, and `every()` verifies that all elements have a length equal to 
+or greater than 5.
+
+After running the code, try changing the predicate of `any()` so it
+returns false:
+
+<?code-excerpt "iterables/test/iterables_test.dart (any-false)"?>
+{% prettify dart %}
+if (items.any((element) => element.contains('Z'))) {
+  print('At least one element contains "Z"');
+} else {
+  print('No element contains "Z"');
+}
+{% endprettify %}
+
+You can also use `any()` to verify that no element of an `Iterable`
+matches a certain condition.
+
+
+### Exercise: Verify that an Iterable satisfies a condition
+
+The following exercise provides practice using the `any()` and `every()` methods, described
+in the previous example. In this case, there's a list of users with the class `User`
+with the member field `age`.
+
+<?code-excerpt "iterables/lib/user.dart"?>
+{% prettify dart %}
+class User {
+  String name;
+  int age;
+}
+{% endprettify %}
+
+Use `any()` and `every()` to implement two functions:
+
+* Part 1: Implement `anyUserUnder18()`.
+  * Return `true` if at least one user is 17 or younger.
+* Part 2: Implement `everyUserOver13()`.
+  * Return `true` if all users are 14 or older.
+
+<!-- [//]: https://gist.github.com/76dc5c8644652adf4e2032912d974ac3 -->
+
+<iframe
+src="{{site.dartpad-embed}}?id=76dc5c8644652adf4e2032912d974ac3&theme=dark"
+frameborder="no"
+height="325"
+width="100%" >
+</iframe>
+
+{{site.alert.secondary}}
+  **Quick review:**
+  * Although you can use `for-in` loops to check conditions, there are better ways to do that.
```

Comment:
```suggestion
Although you can use `for-in` loops to check whether elements satisfy conditions, there are better ways to do that.
```

Reply:
It's not an `Iterables` type (no `s`). I'll think about how to phrase this one.